### PR TITLE
[deliver] apple tv privacy policy update from deliver

### PR DESCRIPTION
**Pull request by aniketgaur**
it is the fix for this bug https://github.com/fastlane/fastlane/issues/14537
feature request is https://github.com/fastlane/fastlane/issues/14558

it also resolves this issue https://github.com/fastlane/fastlane/issues/14547 .

### Checklist
- [ ] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [ ] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

File: deliver/lib/deliver/upload_metadata.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class UploadMetadata
     NON_LOCALISED_VERSION_VALUES = [:copyright]
 
     # Localised app details values
-    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url]
+    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url,:apple_tv_privacy_policy]
```
Comment (janpio) [inline deliver/lib/deliver/upload_metadata.rb]:
```suggestion
    LOCALISED_APP_VALUES = [:name, :subtitle, :privacy_url, :apple_tv_privacy_policy]
```

**Comment by janpio** [comment]
Wow, that is all that is needed?

**Comment by aniketgaur** [comment]
Yeah, I tried it worked.

File: deliver/lib/deliver/options.rb [hunk 2]
```diff
@@ -372,6 +372,10 @@ def self.available_options
                                      description: "Metadata: Localised privacy url",
                                      optional: true,
                                      is_string: false),
+        FastlaneCore::ConfigItem.new(key: :apple_tv_privacy_policy,
+                                     description: "Metadata: Localised apple tv privacy policy",
```
Comment (janpio) [inline deliver/lib/deliver/options.rb]: Let's make it more clear that this requires the text, and not an URL:
```suggestion
                                     description: "Metadata: Localised Apple TV privacy policy text",
```

File: deliver/lib/deliver/download_screenshots.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ def self.run(options, path)
     end
 
     def self.download(options, folder_path)
-      v = options[:use_live_version] ? options[:app].live_version : options[:app].latest_version
+      v = options[:use_live_version] ? options[:app].live_version(platform: options[:platform]) : options[:app].latest_version(platform: options[:platform])
```
Comment (giginet) [inline deliver/lib/deliver/download_screenshots.rb]: Could you explain to need this change?

Comment (giginet) [reply deliver/lib/deliver/download_screenshots.rb]: `platform` could be `nil`. https://github.com/fastlane/fastlane/blob/7da852267f38ab7e08ccc58b6cbc31d39dc6089f/deliver/lib/deliver/options.rb#L81
Do you need to care optional?

Comment (aniketgaur) [reply deliver/lib/deliver/download_screenshots.rb]: this was related to this issue https://github.com/fastlane/fastlane/issues/14547 .
when I am giving value for platform as appletvos it is not getting passed. So options[:app].latest_version(platform: options[:platform]) output "v" for platform "ios".

Will it break something else?

Comment (aniketgaur) [reply deliver/lib/deliver/download_screenshots.rb]: I have tested it. it works fine.

**Comment by aniketgaur** [comment]
I signed it!

**Comment by aniketgaur** [comment]
@janpio and @giginet  what else is needed for the merge?

**Comment by janpio** [comment]
#1: See https://github.com/fastlane/fastlane/pull/14569#issuecomment-483126311 (some of your commits are with different credentials)

**Comment by aniketgaur** [comment]
both of them are mine. should I create new pr?

**Comment by janpio** [comment]
You can also add the email address you used there to your CLA.
Or fix the commits and force push on your branch.
Or reset your branch, commit your changes in one swoop from the correct account, and then force push.
Or not force push, but do it all in a new branch and create a new PR for that one.

**Comment by aniketgaur** [comment]
@janpio  and @giginet  can you review it again?
 I have combined all commits in one.

File: deliver/lib/deliver/download_screenshots.rb (hunk 3 above)
Comment (janpio) [reply deliver/lib/deliver/download_screenshots.rb]: @giginet meant that `options[:platform]` could be nil, thus giving `platform: nil` to `live_version` which might be a problem?

Comment (aniketgaur) [reply deliver/lib/deliver/download_screenshots.rb]: Default argument in live_version is platform:nil. If options [:platform] is nil then that will get pass. which was happening anyways before. Problem was when options [:platform] is not nil and it did not get pass in live_version method call.

**Review by joshdholtz — APPROVED** [review]
This is so 💯 Thanks for doing all of these fixes ❤️

**Pull request merged**